File: backend/backend/src/main/java/com/moovie/backend/domain/usecase/movies/GetMovieDetail.java
package com.moovie.backend.domain.usecase.movies;

import com.moovie.backend.domain.model.MovieDetail;
import com.moovie.backend.domain.repository.MoviesRepository;
import org.springframework.stereotype.Service;

@Service
public class GetMovieDetail {

    private final MoviesRepository moviesRepository;

    public GetMovieDetail(MoviesRepository moviesRepository) {
        this.moviesRepository = moviesRepository;
    }

    public MovieDetail invoke(int movieId) {
        return moviesRepository.getMovieDetail(movieId);
    }
}

File: src/main/kotlin/org/mobyle/domain/usecase/movies/GetMovieDetail.kt
package org.mobyle.domain.usecase.movies

import kotlinx.coroutines.runBlocking
import org.mobyle.domain.model.MovieDetail
import org.mobyle.domain.repository.MoviesRepository

class GetMovieDetail(private val repository: MoviesRepository) {
    operator fun invoke(movieId: Int): MovieDetail = runBlocking {
        repository.getMovieDetail(movieId)
    }
}
